intelmq-mailgen

```diff
@@ -145,12 +146,113 @@ def escape_sql_identifier(ident):
 
 
 def load_events(cur, columns, event_ids):
-    sql_columns = ", ".join(escape_sql_identifier(col) for col in columns)
-    cur.execute("SELECT {} FROM events WHERE id = ANY (%s)".format(sql_columns),
-                (event_ids,))
+    if columns:
+        sql_columns = ", ".join(escape_sql_identifier(col) for col in columns)
+        cur.execute("SELECT {} FROM events WHERE id = ANY (%s)".format(
+                    sql_columns),(event_ids,))
+    else:
+        cur.execute("SELECT * FROM events WHERE id = ANY (%s)", (event_ids,))
     return cur.fetchall()
 
 
+def create_malware_xarf_report(e, notification, config):
+
+    logger.debug("xarf: creating malware report for event {}".format(e))
+    logger.warn("xarf, malware: Using experimental mapping.")
+
+    params = {
+        'schema_cache' : '/tmp/',
+        'schema_url' : 'http://www.x-arf.org/schema/abuse_malware-attack_0.1.4.json',
+        'reported_from' : config["sender"],
+        'report_id' : "TODO intelmq ID", #TODO
+        'category' : 'abuse',
+        'report_type' : 'malware-attack',
+        'source' : e['source.ip'],
+        'source_type' : 'ip-address',
+        'attachment' : None,
+    }
+
+
+    if 'date' in e:
+        params['date'] = e['time.source'].isoformat(sep=' ')
+    else:
+        params['date'] = "2016" # TODO
+
+    if 'source.url' in e:
+        params['download_link'] = e['source.url']
+
+    # have to set 'attachment' e.g. 'text/plain', if 'evidence' is supplied
+
+    report = Xarf(
+        **params
+        #schema_cache='/tmp/',
+        #schema_url='http://www.x-arf.org/schema/abuse_malware-attack_0.1.4.json',
+        # list of potential parameters (with some examples) for this schema:
+
+        #reported_from=
+        #report_id=
+        #category='abuse',
+        #report_type='malware-attack',
+        #destination_system = "enum":["real-world","honeypot","spamtrap","honeyd","nepenthes"], optional
+        #date='Jan  1 2014 02:13:35 +0100',
+        #date=
+        #source='83.169.54.26',
+        #source=e['source.ip'],
+        #source_type='ip-address', "enum":["ipv4","ipv6","ip-address"]
+        #download_link= , optional
+        #download_port= , optional
+        #malware-md5= , optional
+        #antivirus-result= , optional
+        #antivirus-vendor= requires":"Antivirus-Result"
+        #feedback-link=, optional
+        #attachment=  "enum":["none","text/plain", "message/rfc822"]
+        #attachement=
+        #version= , number, optional
+        #occurrences= , integer, optional
+        #tlp= "enum":["white","green","amber","red"],optional
+
+        #evidence= , optional would become the additional attachment
+    )
```

Comment:
It is documented in the schema file, but not as good as it could we. 
I left the comments in, because they will be helpful when improving this even further
and to easy the work to improve the mapping.